### PR TITLE
Unit tests for pto_detector lambda methods

### PR DESCRIPTION
**Pull request by rahul-bhave**
This PR contains 3 unit tests for following 3 methods from pto_detector lambda.
1.get_message_contents
2.get_is_pto
3.write_message

### CODE REVIEW

File: tests/test_unit_pto_detector_get_is_pto.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import os, sys
+from unittest import mock
+from unittest.mock import patch
+from pto_detector import pto_detector
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+@patch('pto_detector.pto_detector.get_is_pto')
```
Comment (qxf2) [inline tests/test_unit_pto_detector_get_is_pto.py:7]: @rahul-bhave - this should patch requests.post not the get_is_pto method :) .... get_is_pto is making exactly one external call. You need to mock that.

Effectively what you would be doing by mocking out requests.post is to say "If requests.POST works fine, then I can assure you that the method get_is_pto is working right". You should be checking this method by setting the returned score to be both 1 and not 1.

File: tests/test_unit_pto_detector_get_message_contents.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import boto3
+import warnings
+from unittest import mock
+from unittest.mock import patch
+from moto import mock_sqs
+from pto_detector import pto_detector
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+@mock_sqs
+@patch('pto_detector.pto_detector.get_message_contents')
```
Comment (qxf2) [inline tests/test_unit_pto_detector_get_message_contents.py:11]: get_message_contents is self contained. So you do not need a patch. You can simply pass it all sorts of events and make sure that it fails gracefully. If it doesn't fail gracefully, give me those events (as a test) and I can fix the method.

File: tests/test_unit_pto_detector_write_message.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from pto_detector import pto_detector
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+# Added following code to fix deprecation warning
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore",category=DeprecationWarning)
+    import imp
+
+@mock_sqs
+@patch('pto_detector.pto_detector.write_message')
```
Comment (qxf2) [inline tests/test_unit_pto_detector_write_message.py:17]: Same misconception with unit tests here too Rahul. You should not mock your method under test. You should instead mock all the methods that your method under test calls.

**Review by qxf2 — CHANGES_REQUESTED** [review]
@rahul-bhave - please look at my review comments. I feel like you need to redo these tests. I have added a conceptual explanation on Trello.

File: tests/test_unit_pto_detector_get_is_pto.py (hunk 1 above)
Comment (rahul-bhave) [reply tests/test_unit_pto_detector_get_is_pto.py]: I have resolved this comment by mocking out requests library.

Comment (rahul-bhave) [reply tests/test_unit_pto_detector_get_is_pto.py]: @qxf2 , I have resolved your comments also, removed unittest dependency from the tests as discussed in sprint meeting. Can I share this for PR for review now?

File: tests/test_unit_pto_detector_get_message_contents.py (hunk 2 above)
Comment (rahul-bhave) [reply tests/test_unit_pto_detector_get_message_contents.py]: @qxf2 , I have resolved your comments also, removed unittest dependency from the tests as discussed in sprint meeting. Can I share this for PR for review now?

File: tests/test_unit_pto_detector_write_message.py (hunk 3 above)
Comment (rahul-bhave) [reply tests/test_unit_pto_detector_write_message.py]: @qxf2 , I have resolved your comments also, removed unittest dependency from the tests as discussed in sprint meeting. Can I share this for PR for review now?

**Comment by rahul-bhave** [comment]
> @rahul-bhave - please look at my review comments. I feel like you need to redo these tests. I have added a conceptual explanation on Trello.

All the comments have been addressed now, working on the comments got during code review

**Pull request merged**